src/Entity.ts

```diff
@@ -8,92 +7,39 @@ type CamelToSnake<T extends string> = string extends T ? string :
         `${C0 extends "_" ? "" : C0 extends Uppercase<C0> ? "_" : ""}${Lowercase<C0>}${CamelToSnake<R>}` :
         "";
 
-type CamelKeysToSnake<T> = T extends readonly any[] ?
+export type CamelKeysToSnake<T> = T extends readonly any[] ?
     { [K in keyof T]: CamelKeysToSnake<T[K]> } :
     T extends object ? {
         [K in keyof T as CamelToSnake<Extract<K, string>>]: CamelKeysToSnake<T[K]>
     } : T;
 
-type EntityPropsOnly<T> = {
+export type EntityPropsOnly<T extends Entity> = {
     [K in keyof T as T[K] extends Function ? never : K]: T[K] extends Entity
         ? EntityPropsOnly<T[K]>
         : (
             T[K] extends Entity[]
-                ? EntityPropsOnly<T[K][]>
+                ? EntityPropsOnly<T[K][number]>[]
                 : T[K]
         );
 }
 
-export class Entity {
-
-    private static async jsonParseAsync<T extends Entity>(sourceObject: T, jsonObject: Partial<CamelKeysToSnake<EntityPropsOnly<T>>>): Promise<T> {
-        const obj = Entity.jsonParse<T>(sourceObject, jsonObject, true);
-        for (const key in obj) {
-            if (obj.hasOwnProperty(key)) {
-                obj[key] = await obj[key];
-            }
-        }
-        return obj;
-    }
+export type PartialEntityPropsOnly<T extends Entity> = Partial<{
```

Comment:
Could you add a comment to this type, and maybe `EntityPropsOnly` too? In particular, I think what's important to highlight is that they are both recursive, and that we need `PartialEntityPropsOnly` rather than `Partial<EntityPropsOnly>` because it's recursively partial, and not only on the top level.